File: skills/utilities/translator-poc/src/actions/translate.py
import os
from bridges.python.src.sdk.leon import leon
from bridges.python.src.sdk.network import Network
from bridges.python.src.sdk.types import ActionParams


def run(params: ActionParams) -> None:
    """Define the winner"""

    print('params', params)

    target_language = 'French'
    text_to_translate = params['utterance']
    network = Network({'base_url': f'{os.environ["LEON_HOST"]}:{os.environ["LEON_PORT"]}/api/v1'})

    leon.answer({
        'key': 'translate',
        'data': {
            'output': f'Translating "{text_to_translate}" to {target_language}'
        }
    })

File: skills/utilities/translator-poc/src/actions/translate.ts
import type { ActionFunction } from '@sdk/types'
import { leon } from '@sdk/leon'
import { Network } from '@sdk/network'

export const run: ActionFunction = async function (params) {
  const targetLanguage = params.slots.target_language.resolution.value
  const textToTranslate = params.new_utterance
  const network = new Network({
    baseURL: `${process.env['LEON_HOST']}:${process.env['LEON_PORT']}/api/v1`
  })
  const systemPrompt = `You are an AI system that translates a given text to "${targetLanguage}" by auto-detecting the source language. You do not add any context to your response.`
  const prompt = `Text to translate: "${textToTranslate}"`

  /**
   * TODO: create SDK methods to handle request and response for every LLM duty
   */
  const response = await network.request({
    url: '/llm-inference',
    method: 'POST',
    data: {
      dutyType: 'custom',
      input: prompt,
      data: {
        systemPrompt
      }
    }
  })
  const translation = response.data.output

  await leon.answer({
    key: 'translate',
    data: {
      output: translation
    }
  })
}
